sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -39,6 +39,16 @@ class SQLQuerySuite extends QueryTest with BeforeAndAfterAll with SQLTestUtils {
   val sqlContext = TestSQLContext
   import sqlContext.implicits._
 
+  test("SPARK-8010: promote numeric to string") {
+    val df = Seq((1, 1)).toDF("key", "value")
+    df.registerTempTable("src")
+    val queryCaseWhen = sql("select case when true then 1.0 else '1' end from src ")
+    val queryCoalesce = sql("select coalesce(null, 1, '1') from src ")
```

Comment:
Looks like these two test queries are not really good examples. We can determine the data type at compile time since `coalesce` and `case when ...` at here basically represent a constant value. Actually, for these cases, why we use `StringType` as the data type?